x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/components/form.tsx

```diff
@@ -42,15 +53,31 @@ interface ExceptionIpEntry {
 
 export const HostIsolationExceptionsForm: React.FC<{
   exception: CreateExceptionListItemSchema | UpdateExceptionListItemSchema;
+  policies: PolicyData[];
   onError: (error: boolean) => void;
-  onChange: (exception: CreateExceptionListItemSchema | UpdateExceptionListItemSchema) => void;
-}> = memo(({ exception, onError, onChange }) => {
+  onChange: (
+    exception: Partial<CreateExceptionListItemSchema> | Partial<UpdateExceptionListItemSchema>
+  ) => void;
+}> = memo(({ exception, onError, policies, onChange }) => {
   const ipEntry = exception.entries[0] as ExceptionIpEntry;
   const [hasBeenInputNameVisited, setHasBeenInputNameVisited] = useState(false);
   const [hasBeenInputIpVisited, setHasBeenInputIpVisited] = useState(false);
   const [hasNameError, setHasNameError] = useState(!exception.name);
   const [hasIpError, setHasIpError] = useState(!ipEntry.value);
 
+  const [isGlobal, setIsGlobal] = useState(isGlobalPolicyEffected(exception.tags));
+  const [selectedPolicies, setSelectedPolicies] = useState<EffectedPolicySelection>({
+    isGlobal,
```

Comment:
Do we need this duplication? Having this `isGlobal` should be enough right? I think you could get rid of the one in line 68

Reply:
This might look like duplicated logic but it is necessary to preserve the selection in the case the user switches from policy to global and back. Using this additional variable I can keep the previous one populated.

Reply:
but, if: 
`selectedPolicies = { isGlobal: false, selected: ['policyId:X', 'policyId:Y']}`
And you set isGlobal to true:
`selectedPolicies = { isGlobal: true, selected: ['policyId:X', 'policyId:Y']}`

The selected attribute can keep the previous selection right? Is there another reason why you need another `isGlobal` flag?

Reply:
@dasansol92 not exactly.

`selectedPolicies` is set from the data that comes from the `EffectedPoliciesSelector`. `EffectedPoliciesSelector` will return this `selected` structure empty when `isGlobal` comes. This effectively means I will have to code the same logic anyway but in another place.

the logic to keep the current is global and the current selection needs to exists somewhere. I find it easier to simply have a separate variable.

Reply:
After checking a bit further `EffectedPoliciesSelector` is doing the "right" thing and preserving the state of passed values between switches. That means the logic I thought  required was not.

I pushed a commit with the updated logic.